lib/src/git.rs

```diff
@@ -1322,11 +1322,21 @@ fn push_refs(
         }
     })?;
     let mut remaining_remote_refs: HashSet<_> = qualified_remote_refs.iter().copied().collect();
+    let mut remote_message: Option<String> = None;
     let mut push_options = git2::PushOptions::new();
     let mut proxy_options = git2::ProxyOptions::new();
     proxy_options.auto();
     push_options.proxy_options(proxy_options);
     let mut callbacks = callbacks.into_git();
+    callbacks.sideband_progress(|progress| {
+        let message = String::from_utf8_lossy(progress);
+        if let Some(remote_message) = &mut remote_message {
+            remote_message.push_str(&message);
+        } else {
+            remote_message = Some(message.to_string());
+        }
```

Comment:
I think this can be written like this:
```rs
remote_message.get_or_insert_with(String::new()).push_str(&message);
```

But maybe we can even replace the `Option<String>` by a `String`? It seems unlikely that the caller will care about the difference between a message that's an empty string and no message at all (if the Git protocol even allows the remote to send an empty string).

Reply:
Oh, the next commit undoes most of this anyway. I would suggest squashing them.